run/image-processing/image.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+// [START run_imageproc_handler_setup]
+const gm = require('gm').subClass({imageMagick: true});
+const fs = require('fs');
+const {promisify} = require('util');
+const path = require('path');
+const {Storage} = require('@google-cloud/storage');
+const storage = new Storage();
+const vision = require('@google-cloud/vision').v1p1beta1;
+
+const client = new vision.ImageAnnotatorClient();
+
+const {BLURRED_BUCKET_NAME} = process.env;
+// [END run_imageproc_handler_setup]
+
+// [START run_imageproc_handler_analyze]
+// Blurs uploaded images that are flagged as Adult or Violence.
+exports.blurOffensiveImages = async event => {
+  // This event represents the triggering Cloud Storage object.
```

Comment:
Why not use `object` in the function signature instead of reassigning it? Looking at the call site, we use `data`. Maybe change the parameter to `data` or `imageData`?

Reply:
Maybe `data` isn't the greatest name. But I'd not re-assign immediately.